r2-testapp/src/main/java/org/readium/r2/testapp/audiobook/AudiobookActivity.kt

```diff
@@ -1,151 +1,109 @@
-        menuDrm = menu?.findItem(R.id.drm)
-        menuToc = menu?.findItem(R.id.toc)
-        menuBmk = menu?.findItem(R.id.bookmark)
-        menuSettings = menu?.findItem(R.id.settings)
-
-        menuSettings?.isVisible = false
-        menuDrm?.isVisible = false
-        return true
+    override fun onStart() {
+        super.onStart()
+        updateActivityTitle()
     }
 
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        when (item.itemId) {
 
-            R.id.toc -> {
-                val intent = Intent(this, R2OutlineActivity::class.java).apply {
-                    putPublicationFrom(this@AudiobookActivity)
-                    putExtra("bookId", bookId)
-                }
-                startActivityForResult(intent, 2)
-                return true
-            }
-            R.id.settings -> {
-                // TODO do we need any settings ?
-                return true
-            }
-            R.id.bookmark -> {
-                val locator = currentLocator.value
-
-                val bookmark = Bookmark(bookId, publicationIdentifier, resourceIndex = currentResource.toLong(), locator = locator)
-                bookmarksDB.bookmarks.insert(bookmark)?.let {
-                    launch {
-                        toast("Bookmark added")
-                    }
-                } ?: run {
-                    launch {
-                        toast("Bookmark already exists")
-                    }
-                }
-
-                return true
-            }
-            R.id.drm -> {
-                startActivityForResult(intentFor<DRMManagementActivity>("publication" to publicationPath), 1)
-                return true
-            }
-
-            else -> return false
+    private fun updateActivityTitle() {
+        title = when (supportFragmentManager.fragments.last()) {
+            is OutlineFragment -> publication.metadata.title
+            is DrmManagementFragment -> getString(R.string.title_fragment_drm_management)
+            else -> null
         }
+    }
 
+    override fun getDefaultViewModelProviderFactory(): ViewModelProvider.Factory {
+        return modelFactory
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
-        activitiesLaunched.getAndDecrement()
+    private fun handleReaderFragmentEvent(event: ReaderViewModel.Event) {
+        when(event) {
+            is ReaderViewModel.Event.OpenOutlineRequested -> showOutlineFragment()
+            is ReaderViewModel.Event.OpenDrmManagementRequested -> showDrmManagementFragment()
+        }
     }
 
-    override fun onStop() {
-        super.onStop()
-        progressDialog.dismiss()
+    private fun showOutlineFragment() {
+        supportFragmentManager.beginTransaction()
+            .add(R.id.activity_container, OutlineFragment::class.java, Bundle(), ReaderActivity.OUTLINE_FRAGMENT_TAG)
+            .hide(readerFragment)
+            .addToBackStack(null)
+            .commit()
```

Comment:
Fragment transactions would look nicer and safer with the `commit` API.

```suggestion
        supportFragmentManager.commit {
            add(R.id.activity_container, OutlineFragment::class.java, Bundle(), ReaderActivity.OUTLINE_FRAGMENT_TAG)
            hide(readerFragment)
            addToBackStack(null)
        }
```